File: android/ValleCOM/app/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorSugerencias.java
package com.valleapp.vallecom.adaptadores;

import android.content.Context;
import android.util.TypedValue;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.List;

import com.valleapp.vallecom.R;

/**
 * Created by valle on 17/09/14.
 */
public class AdaptadorSugerencias extends ArrayAdapter<JSONObject> {

    Context cx;
    List<JSONObject> values;

    public AdaptadorSugerencias(Context context, List<JSONObject> obj) {
        super(context, R.layout.linea_simple, obj);
        this.cx = context;
        this.values= obj;

    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        LayoutInflater inflater = (LayoutInflater) cx
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View rowView = inflater.inflate(R.layout.linea_simple, parent, false);

        try {
            TextView can = (TextView) rowView.findViewById(R.id.labelTitle);
            // Set the text size 25 dip for ListView each item
            can.setTextSize(TypedValue.COMPLEX_UNIT_DIP,25);

            String sug = values.get(position).getString("sugerencia");
            can.setText(String.format("%s", sug));
            RelativeLayout btnInfo = (RelativeLayout)rowView.findViewById(R.id.item);
            btnInfo.setTag(sug);

        } catch (JSONException e) {
            e.printStackTrace();
        }


        return rowView;
    }

}

File: android/ValleTPV/ValleCOM/src/main/java/com/valleapp/vallecom/adaptadores/AdaptadorSugerencias.kt
package com.valleapp.vallecom.adaptadores

import android.content.Context
import android.graphics.Color
import android.util.TypedValue
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.LinearLayout
import android.widget.TextView
import com.valleapp.valletpv.R
import com.valleapp.vallecom.activitys.Sugerencias
import org.json.JSONException
import org.json.JSONObject
import androidx.core.graphics.toColorInt

class AdaptadorSugerencias(
    private val context: Context,
    private val values: List<JSONObject>
) : BaseAdapter() {

    override fun getCount(): Int {
        return values.size
    }

    override fun getItem(position: Int): JSONObject {
        return values[position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup?): View {
        val view = convertView ?: LayoutInflater.from(context).inflate(R.layout.item_sugerencia, parent, false)
        val tituloTextView: TextView = view.findViewById(R.id.txtTitulo)
        val precioTextView: TextView = view.findViewById(R.id.txtPrecio)
        val itemLayout: LinearLayout = view.findViewById(R.id.item)

        try {
            val item = getItem(position)
            val sugerencia = item.getString("sugerencia")
            val incremento = item.getDouble("incremento")

            tituloTextView.setTextSize(TypedValue.COMPLEX_UNIT_DIP, 25f)
            tituloTextView.text = sugerencia

            // Mostrar precio si incremento > 0
            if (incremento > 0) {
                precioTextView.visibility = View.VISIBLE
                "+%.2f€".format(incremento).also { precioTextView.text = it }
                precioTextView.setTextSize(TypedValue.COMPLEX_UNIT_DIP, 25f)
                // Cambiar color cuando hay incremento
                itemLayout.setBackgroundColor("#E8F5E8".toColorInt())
                tituloTextView.setTextColor("#2E7D32".toColorInt())
            } else {
                precioTextView.visibility = View.GONE
                itemLayout.setBackgroundColor(Color.TRANSPARENT)
                tituloTextView.setTextColor(Color.BLACK)
            }

            itemLayout.tag = item

            // Agregar onClick listener
            itemLayout.setOnClickListener {
                if (context is Sugerencias) {
                    context.onClickItemSimple(it)
                }
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return view
    }
}
